code/src/test/kotlin/lexicalTests/LexerTest.kt

```diff
@@ -76,6 +74,176 @@ class LexerTest {
+                { token -> token is lexer.Token.Type.Func },
+                0, 0)
+        assertPositionalToken(tokens[1],
+                { token -> token is lexer.Token.Identifier && token.value == "print" },
+                5, 0)
+        assertPositionalToken(tokens[2],
+                { token -> token is lexer.Token.SpecialChar.Equals },
+                11, 0)
+        assertPositionalToken(tokens[3],
+                { token -> token is lexer.Token.SpecialChar.ParenthesesStart },
+                13, 0)
+        assertPositionalToken(tokens[4],
+                { token -> token is lexer.Token.Type.Text },
+                14, 0)
+        assertPositionalToken(tokens[5],
+                { token -> token is lexer.Token.Identifier && token.value == "message" },
+                18, 0)
+        assertPositionalToken(tokens[6],
+                { token -> token is lexer.Token.SpecialChar.ParenthesesEnd },
+                25, 0)
+        assertPositionalToken(tokens[7],
+                { token -> token is lexer.Token.SpecialChar.Colon },
+                27, 0)
+        assertPositionalToken(tokens[8],
+                { token -> token is lexer.Token.Type.None },
+                29, 0)
+        assertPositionalToken(tokens[9],
+                { token -> token is lexer.Token.SpecialChar.EndOfLine },
+                33, 0)
+        assertPositionalToken(tokens[10],
+                { token -> token is lexer.Token.SpecialChar.BlockStart },
+                0, 1)
+        assertPositionalToken(tokens[11],
+                { token -> token is lexer.Token.SpecialChar.BlockEnd },
+                2, 1)
+        assertPositionalToken(tokens[12],
+                { token -> token is lexer.Token.SpecialChar.EndOfLine },
+                3, 1)
+    }
+
+    @org.junit.jupiter.api.Test
+    fun lexerTestDoesntRequireWhitespaces() {
+        val lex = Lexer("x=y[0]")
+
+        val tokens = lex.getTokenSequence().toList()
+
+        assertPositionalToken(tokens[0],
+                { token -> token is lexer.Token.Identifier && token.value == "x" },
+                0, 0)
+
+        assertPositionalToken(tokens[1],
+                { token -> token is lexer.Token.SpecialChar.Equals },
+                1, 0)
+
+        assertPositionalToken(tokens[2],
+                { token -> token is lexer.Token.Identifier && token.value == "y" },
+                2, 0)
+
+        assertPositionalToken(tokens[3],
+                { token -> token is lexer.Token.SpecialChar.SquareBracketStart },
+                3, 0)
+
+        assertPositionalToken(tokens[4],
+                { token -> token is lexer.Token.Literal.Number && token.value == 0.0 },
+                4, 0)
+
+        assertPositionalToken(tokens[5],
+                { token -> token is lexer.Token.SpecialChar.SquareBracketEnd },
+                5, 0)
+    }
+
+    @org.junit.jupiter.api.Test
+    fun lexerTestInlineLambdaExpression() {
+        val lex = Lexer("x = 1 to 10 map {value * 2}")
```

Comment:
is this test any different that the func assignment? this seems more relevant in the parser - this doesn't really have any "unique" cases that the others did not cover. So it's just going to make building slower without adding anything new. maybe just have a tests for ALL symbols (as you already have for all types)